File: Second Year/PDOO/Napakalakirb/lib/Monster.rb
# encoding: UTF-8
#
#   Programación y diseño orientado a objetos
#   Grado en Ingeniería Informática
#
#  Ángela Castro Jimenez
#  Francisco Carrillo Pérez
module Napakalaki
require_relative 'Prize'
require_relative 'BadConsequence'
require_relative 'player'
  class Monster < Card
      attr_accessor :name, :combatLevel, :badConsequence,:prize,:levelChangeAgainstCultistPlayer

    def initialize (name, combatLevel, badConsequence,prize,levels)
      @name = name
      @combatLevel = combatLevel
      @prize = prize
      @badConsequence = badConsequence
      @levelChangeAgainstCultistPlayer = levels
      
    end
    
    
    def getLevelsGained()
      return @prize.getLevel
    end
    
    def getTreasuresGained()
      return @prize.getTreasures
    end
    
    def kills()
      return @badConsequence.myBadConsequenceIsDeath
    end
    def getCombatLevel
     @combatLevel
   end
   def getBadConsequence
      return @badConsequence
   end
    def to_s 
        "Nombre: #{@name} \n Nivel: #{@combatLevel} \n Premio: #{@prize.to_s}
        \n Mal rollo: #{@badConsequence.to_s}"
    end

    def basic_value
        return getCombatlevel;
    end

    def special_value
      return @combatLevel + @levelChangeAgainstCultistPlayer;
    end

    
  end
end

File: Second Year/PDOO/Napakalaki/src/NapakalakiGame/Monster.java
/*  
 *  Programación y diseño orientado a objetos
 *  Grado en Ingeniería Informática
 * 
 *  
 *
 *  Angela Castro Jiménez
 *  Francisco Carrillo Pérez
 */
package NapakalakiGame;

public class Monster implements Card {
    private String name;
    private int combatLevel;
    private int levelChangedAgainstCultistPlayer;
    private Prize prize = new Prize();
    private BadConsequence badConsequence = new BadConsequence();
    
    private String tipo;
    
    public Monster(){
        name = "";
        combatLevel = 0;
        
    }
     
    public Monster(String n, int l){
        name = n;
        combatLevel = l;
    }

    public Monster(String n, int l, BadConsequence b, Prize p){
        name = n;
        combatLevel = l;
        badConsequence = b;
        prize = p;
        
    }
    
    public Monster(String n, int l, BadConsequence b, Prize p, int levels){
        name = n;
        combatLevel = l;
        badConsequence = b;
        prize = p;
        levelChangedAgainstCultistPlayer = levels;
        
    }
    
    public String getTipo(){
        return tipo;
    }
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;  
    }
    
    public Prize getPrize(){
         return prize;
    }
    
    public BadConsequence getBadConsequence(){
         return badConsequence;
    }
    
    public int getLevelsGained(){
        return prize.getLevels();
        
    }
    
     public int getTreasuresGained(){
        return prize.getTreasures();
    }
    
    public boolean kills(){
        return badConsequence.myBadConsequenceIsDeath();
    }
     
     private int levelChangedAgainstCultistPlayer()
     {
         return levelChangedAgainstCultistPlayer;
     }
     
     @Override
     public int getBasicValue()
     {
         return combatLevel;
         
     }
    
    @Override
    public int getSpecialValue() {
        return this.getCombatLevel() + this.levelChangedAgainstCultistPlayer();
    }
    
    public String toString(){
        return "Nombre: "+name
                +".\nNivel: " + Integer.toString(combatLevel)
                +".\nPremio: " + prize.toString()
                +"\n"+ badConsequence.toString();
        }
    
    
        
   
    }
    /*PRUEBA
        public static void main(String args[]){
          Prize prize = new Prize(4,2);
          BadConsequence malrollo = new BadConsequence("Pierdes 5 niveles", 5, 0, 0);
          Monster monstruo = new Monster("El monstruo de prueba", 5, malrollo, prize);
          
           System.out.println(monstruo.getName());
           System.out.println(monstruo.getCombatLevel()); 
           System.out.println(monstruo.getPrize()); 
           System.out.println(monstruo.getMalrollo()); 
           System.out.println(monstruo.toString());    
}
        }*/
     

